paseos/attitude/disturbance_calculations.py

```diff
@@ -1,26 +1,120 @@
-    T = [0, 0, 0]
-    return np.array(T)
-
-
-def calculate_magnetic_torque():
-    # calculations for torques
-    # T must be in actor body fixed frame (to be discussed)
-    T = [0, 0, 0]
-    return np.array(T)
+# this functions could be implemented inside the attitude model class but I'd rather have it
+# separately right now.
+
+# OUTPUT NUMPY ARRAYS
+import numpy as np
+import trimesh
+
+from paseos.utils.reference_frame_transfer import transformation_matrix_eci_rpy, transformation_matrix_rpy_body, \
+   eci_to_rpy
+
+
+
+def calculate_aero_torque(r,v,mesh,temperature_spacecraft, transformation_matrix_rpy_body,):
+    """ Calculates the aerodynamic torque on the satellite. Possible further inputs are temperature_spacecraft from the
+    thermal model.
+     Returns:
+         T: torque vector in the spacecraft body frame
+    """
+
+    #  Set data useful for aerodynamic coefficients calculation
+    temperature_spacecraft = 300  # [K]  we could connect this with the thermal model
+    temperature_gas = 1000  # [K]  valid for upper stratosphere, could be completely wrong higher up
+    altitude = np.linalg.norm(r)-6371  # [km]
+    density = 10**(-(altitude+1285)/151)  # from data fit from thermospheric mass density
+    R = 8.314462  # [J/(K mol)] Universal Gas Constant
+    molecular_speed_ratio_t = np.linalg.norm(v)/np.sqrt(2*R*temperature_gas)  # Used in the Cd and Cl calculation
+    molecular_speed_ratio_r = np.linalg.norm(v)/np.sqrt(2*R*temperature_spacecraft)  # Used in the Cd and Cl calculation
+    accommodation_parameter = 0.85
+
+    #  Get the normal vectors of all the faces of the mesh in the spacecraft body reference frame, and then they get
+    #  translated in the Roll Pitch Yaw frame with a transformation from paseos.utils.reference_frame_transfer.py
+    face_normals_sbf = mesh.face_normals[0:12]
+    face_normals_rpy = np.dot(transformation_matrix_rpy_body, face_normals_sbf.T).T  # Check  transposition
+
+    #  Get the velocity and transform it in the Roll Pitch Yaw frame. Get the unit vector associated with the latter
+    v_rpy = eci_to_rpy(v, r, v)
+    unit_v_rpy = v_rpy/np.linalg.norm(v_rpy)
+
+    #  Get the normals, the areas, the angle of attack and the centroids of the faces with airflow, confronting them
+    #  with the velocity vector direction. If the dot product between velocity and normal is positive, the plate
+    #  corresponding to the normal is receiving airflow.
+
+    #  Initialize parameters and variables for the for loop.
+    j = 0
+    normals_faces_with_airflow = np.zeros((6, 3))  # Maximum six faces have airflow
+    alpha = [0, 0, 0, 0, 0, 0]  # angle of attack matrix, stores the angles of attack in radiant
+    area_all_faces_mesh = mesh.area_faces  # ordered list of all faces' areas
+    area_faces_airflow = [0, 0, 0, 0, 0, 0]
+    centroids_faces_airflow = [0, 0, 0, 0, 0, 0]
+
+    for i in range(12):
+        if np.dot(face_normals_rpy[i], v_rpy) > 0:

```

Comment:
This is not really clear. Can you comment?

Reply:
commented on the updated code (the comment is above the initialization of the parameters)